src/systems/filecoin_mining/storage_mining/storage_mining_subsystem.go

```diff
@@ -78,40 +134,39 @@ func (sms *StorageMiningSubsystem_I) _tryLeaderElection() {
 	candidates := sms.StorageProving().Impl().GenerateElectionPoStCandidates(postRandomness, provingSet)
 
 	if len(candidates) <= 0 {
-		return // fail to generate post candidates
+		return nil // fail to generate post candidates
 	}
 
-	// TODO Fix
-	util.TODO()
-	var currState stateTree.StateTree
 	winningCandidates := make([]sector.PoStCandidate, 0)
-	st := sms._getStorageMinerActorState(currState, sms._keyStore().MinerAddress())
 
-	numMinerSectors := uint64(len(st.SectorTable().Impl().ActiveSectors_.SectorsOn()))
+	numMinerSectors := uint64(len(sma.SectorTable().Impl().ActiveSectors_.SectorsOn()))
 	for _, candidate := range candidates {
 		sectorNum := candidate.SectorID().Number()
-		sectorPower, ok := st._getSectorPower(sectorNum)
+		sectorPower, ok := sma._getSectorPower(sectorNum)
 		if !ok {
 			// panic(err)
-			return
+			return nil
 		}
 		if sms._consensus().IsWinningPartialTicket(currState, candidate.PartialTicket(), sectorPower, numMinerSectors) {
 			winningCandidates = append(winningCandidates, candidate)
 		}
 	}
 
 	if len(winningCandidates) <= 0 {
-		return
+		return nil
 	}
 
-	// Randomness for ticket generation in block production
-	randomness1 := sms._consensus().GetTicketProductionRand(sms._blockchain().BestChain(), sms._blockchain().LatestEpoch())
-	newTicket := sms.PrepareNewTicket(randomness1, sms._keyStore().MinerAddress())
-
 	postProof := sms.StorageProving().Impl().CreateElectionPoStProof(postRandomness, winningCandidates)
-	chainHead := sms._blockchain().BestChain().HeadTipset()
 
-	sms._blockProducer().GenerateBlock(postProof, winningCandidates, newTicket, chainHead, sms._keyStore().MinerAddress())
+	var ctc sector.ChallengeTicketsCommitment // TODO: proofs to fix when complete
```

Comment:
@porcuquine will need to touch this up once we decide what to do with CommT for `ePoSt` and `sPoSt` fyi